app/src/main/java/de/westnordost/streetcomplete/util/ktx/LocalDate.kt

```diff
@@ -1,19 +1,48 @@
+@file:Suppress("NOTHING_TO_INLINE")
+
 package de.westnordost.streetcomplete.util.ktx
 
-import java.time.Instant
-import java.time.LocalDate
-import java.time.Month
-import java.time.ZoneId
+import kotlinx.datetime.Clock
+import kotlinx.datetime.DateTimeUnit
+import kotlinx.datetime.Instant
+import kotlinx.datetime.LocalDate
+import kotlinx.datetime.LocalDateTime
+import kotlinx.datetime.Month
+import kotlinx.datetime.TimeZone
+import kotlinx.datetime.atStartOfDayIn
+import kotlinx.datetime.minus
+import kotlinx.datetime.plus
+import kotlinx.datetime.toInstant
+import kotlinx.datetime.toLocalDateTime
 
 fun LocalDate.toInstant(): Instant =
-    this.atStartOfDay(ZoneId.systemDefault()).toInstant()
+    this.atStartOfDayIn(TimeZone.currentSystemDefault())
 
-fun LocalDate.toEpochMilli(): Long = this.toInstant().toEpochMilli()
+fun LocalDate.toEpochMilli(): Long = this.toInstant().toEpochMilliseconds()
 
 fun Instant.toLocalDate(): LocalDate =
-    this.atZone(ZoneId.systemDefault()).toLocalDate()
+    this.toLocalDateTime(TimeZone.currentSystemDefault()).date
 
 fun isApril1st(): Boolean {
-    val now = LocalDate.now()
+    val now = systemTimeNow().toLocalDate()
```

Comment:
Worth making this into an extension function like you did for `LocalDateTime.Companion.now()` some lines below? I have seen a few such usages IIRC during review.

Reply:
`systemTimeNow()` is called 8 times, 6 of those are `systemTimeNow().toLocalDate()`.

Reply:
Ok, was just a suggestion.